test/main.test.ts

```diff
@@ -0,0 +1,143 @@
+import {
+    parseFile,
+    parseStr
+} from './test_util'
+import {
+    ExampleSyntaxError
+} from "../src/index"
+
+test('Common | An empty string', () => {
+    let parsed = parseStr('')
+    let expected = {}
+    expect(parsed).toStrictEqual(expected)
+})
+
+test('Rust / C | Code section', () => {
+    let parsed = parseFile('./test/samples/code_section.rs')
+    let expected = {
+        RustStyleFragment: 'println!("A normal Rust-style fragment");'
+    }
+    expect(parsed).toStrictEqual(expected)
```

Comment:
Not a critical comment, but I'd like to mention that Jest (and Vitest) has the [Snapshot Testing](https://jestjs.io/docs/snapshot-testing) feature. It might be useful here. Especially the `toMatchInlineSnapshot()` assertion.